patch file
diff --git a/dot_buildrootconfig b/dot_buildrootconfig
new file mode 100644
index 0000000000..e69de29bb2
diff --git a/fs/cpio/init b/fs/cpio/init
index f74ef7e15f..df2ba4d75d 100755
--- a/fs/cpio/init
+++ b/fs/cpio/init
@@ -13,3 +13,6 @@ if (exec 0</dev/console) 2>/dev/null; then
 fi
 
 exec /sbin/init "$@"
+/bin/busybox echo "fell off end of /sbin/init"
+/bin/busybox echo "powering off..."
+/bin/busybox poweroff -f
