From c12549d31641aa565f8fc773ec28c47114dcd50e Mon Sep 17 00:00:00 2001
From: root <root@localhost.localdomain>
Date: Wed, 2 Oct 2024 12:35:54 +0000
Subject: [PATCH 1/1] c

---
 fs/cpio/init | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/fs/cpio/init b/fs/cpio/init
index f74ef7e15f..df2ba4d75d 100755
--- a/fs/cpio/init
+++ b/fs/cpio/init
@@ -13,3 +13,6 @@ if (exec 0</dev/console) 2>/dev/null; then
 fi
 
 exec /sbin/init "$@"
+/bin/busybox echo "fell off end of /sbin/init"
+/bin/busybox echo "powering off..."
+/bin/busybox poweroff -f
-- 
2.46.0

