diff --git a/app/build.gradle b/app/build.gradle
index fca317f1..e515fb74 100755
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -1,12 +1,19 @@
 apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-android-extensions'
 
 android { // main settings for your application
-    compileSdkVersion = 29
-    buildToolsVersion = '29.0.3'
+    compileSdkVersion 29
+    buildToolsVersion '29.0.3'
     ndkVersion '21.4.7075529'
 
+    packagingOptions {
+        exclude "**/module-info.class"
+    }
+
+    viewBinding {
+        enabled true
+    }
+
     defaultConfig {
         applicationId "media.player.pro"
         minSdkVersion 26
@@ -14,6 +21,8 @@ android { // main settings for your application
         versionCode 1
         versionName "1.0"
 
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+
         externalNativeBuild {
             cmake {
                 cppFlags "-std=c++17 -DANDROID_STL=libc++_shared"
@@ -29,6 +38,13 @@ android { // main settings for your application
         }
     }
 
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+    }
+
     sourceSets {
         main {
             jniLibs {
@@ -52,15 +68,16 @@ dependencies {
     implementation fileTree(include: ['*.jar'], dir: 'libs')
     implementation 'com.android.support:appcompat-v7:28.0.0'
     implementation 'com.android.support.constraint:constraint-layout:1.1.3'
-    implementation 'androidx.core:core-ktx:1.2.0-alpha02'
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    implementation 'com.jraska:console:1.0.0'
+    implementation 'androidx.core:core-ktx:1.6.0'
+    implementation 'com.jraska:console:1.2.0'
     implementation "commons-io:commons-io:2.6"
     implementation "androidx.annotation:annotation:1.1.0"
     implementation "androidx.drawerlayout:drawerlayout:1.0.0"
     implementation "androidx.preference:preference:1.1.1"
     implementation "androidx.viewpager:viewpager:1.0.0"
-
+    testImplementation 'junit:junit:4.13.2'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
 }
 
 repositories {
diff --git a/app/src/main/java/media/player/pro/MainActivity.kt b/app/src/main/java/media/player/pro/MainActivity.kt
index 74a362bc..d93b8008 100644
--- a/app/src/main/java/media/player/pro/MainActivity.kt
+++ b/app/src/main/java/media/player/pro/MainActivity.kt
@@ -9,7 +9,6 @@ import android.view.MenuItem
 import android.view.View
 import android.widget.Button
 import androidx.appcompat.app.AppCompatActivity
-import kotlinx.android.synthetic.main.slide_out_terminal.*
 import liblayout.Builder
 import liblayout.UiThread
 import libmedia.Media
@@ -28,8 +27,6 @@ class MainActivity : AppCompatActivity() {
         // set log view
         setContentView(R.layout.slide_out_terminal)
 
-        // obtain log view instance
-
         // obtain log view instance
         terminalController = TerminalController()
         terminalController!!.onCreate(this, terminal_view)
@@ -46,7 +43,6 @@ class MainActivity : AppCompatActivity() {
         media = Media(this).also {
             it.init()
             it.loadMediaAssetAsFile("00001313_48000.raw")
-//            it.loadMediaAssetAsFile("FUNKY_HOUSE.raw")
             it.loop(true)
         }
 
@@ -169,4 +165,4 @@ class MainActivity : AppCompatActivity() {
     override fun onBackPressed() {
         if (!terminalController!!.onBackPressed()) super.onBackPressed()
     }
-}
\ No newline at end of file
+}
diff --git a/build.gradle b/build.gradle
index df8fe1b9..ee332e6f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -1,14 +1,14 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
 buildscript {
-    ext.kotlin_version = '1.3.41'
+    ext.kotlin_version = '1.5.10'
     repositories {
         google()
         jcenter()
-        
+        mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.1'
+        classpath 'com.android.tools.build:gradle:3.6.1'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
@@ -19,7 +19,7 @@ allprojects {
     repositories {
         google()
         jcenter()
-        
+        mavenCentral()
     }
 }
 
diff --git a/gradle/wrapper/gradle-wrapper.properties b/gradle/wrapper/gradle-wrapper.properties
index 5acc58f1..cdf93caf 100644
--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.1-all.zip
