diff --git a/app/build.gradle b/app/build.gradle
index fca317f1..ed83dd74 100755
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -5,7 +5,7 @@ apply plugin: 'kotlin-android-extensions'
 android { // main settings for your application
     compileSdkVersion = 29
     buildToolsVersion = '29.0.3'
-    ndkVersion '21.4.7075529'
+    ndkVersion '27.1.12297006'
 
     defaultConfig {
         applicationId "media.player.pro"
@@ -54,7 +54,7 @@ dependencies {
     implementation 'com.android.support.constraint:constraint-layout:1.1.3'
     implementation 'androidx.core:core-ktx:1.2.0-alpha02'
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    implementation 'com.jraska:console:1.0.0'
+    //implementation 'com.jraska:console:1.0.0'
     implementation "commons-io:commons-io:2.6"
     implementation "androidx.annotation:annotation:1.1.0"
     implementation "androidx.drawerlayout:drawerlayout:1.0.0"
diff --git a/build.gradle b/build.gradle
index df8fe1b9..4fda6146 100644
--- a/build.gradle
+++ b/build.gradle
@@ -1,18 +1,8 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
-
-buildscript {
-    ext.kotlin_version = '1.3.41'
-    repositories {
-        google()
-        jcenter()
-        
-    }
-    dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.1'
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
-        // NOTE: Do not place your application dependencies here; they belong
-        // in the individual module build.gradle files
-    }
+plugins {
+    id 'com.android.application' version '7.2.2' apply false
+    id 'com.android.library' version '7.2.2' apply false
+    id 'org.jetbrains.kotlin.android' version '1.6.10' apply false
 }
 
 allprojects {
