diff --git a/build.gradle b/build.gradle
index df8fe1b9..66d08ca8 100644
--- a/build.gradle
+++ b/build.gradle
@@ -5,7 +5,7 @@ buildscript {
     repositories {
         google()
         jcenter()
-        
+        maven()
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:3.5.1'
@@ -19,7 +19,7 @@ allprojects {
     repositories {
         google()
         jcenter()
-        
+        maven()
     }
 }
 
