From dda357ad379a897bba6523fe68af2b3fc860fdcd Mon Sep 17 00:00:00 2001
From: root <root@localhost.localdomain>
Date: Tue, 1 Oct 2024 09:45:06 +0000
Subject: [PATCH 1/1] c

---
 CMakeLists.txt | 38 ++++++++++++++++++++++++++++++++++++++
 1 file changed, 38 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 00000000..d1690a62
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,38 @@
+cmake_minimum_required(VERSION 3.4.1)
+
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/virglrenderer.cross.build "
+[binaries]
+ar = '${CMAKE_ANDROID_NDK}/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android-ar'
+c = ['${CMAKE_ANDROID_NDK}/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android29-clang']
+cpp = ['${CMAKE_ANDROID_NDK}/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android29-clang++', '-fno-exceptions', '-fno-unwind-tables', '-fno-asynchronous-unwind-tables', '-static-libstdc++']
+c_ld = 'lld'
+cpp_ld = 'lld'
+strip = '${CMAKE_ANDROID_NDK}/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android-strip'
+# Android doesn't come with a pkg-config, but we need one for Meson to be happy not
+# finding all the optional deps it looks for.  Use system pkg-config pointing at a
+# directory we get to populate with any .pc files we want to add for Android
+pkg-config = ['env', 'PKG_CONFIG_LIBDIR=${CMAKE_ANDROID_NDK}/pkgconfig', '/usr/bin/pkg-config']
+
+[host_machine]
+system = 'android'
+cpu_family = 'aarch64'
+cpu = 'armv8'
+endian = 'little'
+")
+
+include(ExternalProject)
+
+ExternalProject_Add(proj_virglrenderer
+    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/virglrenderer-build
+    SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}
+    BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/virglrenderer-build
+    CONFIGURE_COMMAND meson setup ${CMAKE_CURRENT_BINARY_DIR}/virglrenderer-build ${CMAKE_CURRENT_SOURCE_DIR} --cross-file ${CMAKE_CURRENT_BINARY_DIR}/virglrenderer.cross.build --prefix ${CMAKE_CURRENT_BINARY_DIR}/virglrenderer-install
+    BUILD_COMMAND meson compile -C ${CMAKE_CURRENT_BINARY_DIR}/virglrenderer-build
+    INSTALL_COMMAND meson install -C ${CMAKE_CURRENT_BINARY_DIR}/virglrenderer-build
+    BUILD_BYPRODUCTS ${CMAKE_CURRENT_BINARY_DIR}/virglrenderer-install/lib/libvirglrenderer.a
+)
+
+add_library(virglrenderer INTERFACE)
+add_dependencies(virglrenderer proj_virglrenderer)
+target_include_directories(virglrenderer INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/virglrenderer-install/include)
+target_link_libraries(virglrenderer INTERFACE virglrenderer-install/lib/libvirglrenderer.a)
-- 
2.46.0

